pkg-manager/core/test/install/local.ts

```diff
@@ -240,6 +241,66 @@ test('update tarball local package when its integrity changes', async () => {
   expect(manifestOfTarballDep.dependencies['is-positive']).toBe('^2.0.0')
 })
 
+// Similar to the test above, but for a filtered install.
+// Regression test for https://github.com/pnpm/pnpm/pull/9805.
+test('update tarball local package when its integrity changes (filtered install)', async () => {
```

Comment:
Here's a screenshot of this test run before the fix in this PR:

<img width="842" height="1162" alt="Image" src="https://github.com/user-attachments/assets/607e16cf-b87b-4923-9ea6-0878af0fc33b" />